pkg/controllers/user/pipeline/controller/pipelineexecution/statesyncer.go

```diff
@@ -48,7 +49,15 @@ func (s *ExecutionStateSyncer) syncState() {
 		return
 	}
 	if err := s.pipelineEngine.PreCheck(); err != nil {
+		//fail to connect engine, mark the remaining executions as failed
```

Comment:
What about adding a new state `Unknown`?

Reply:
I think an `Unknow` state of a pipeline execution is confusing for the users. They should know and check any unexpected failure in the CI procedure.